### PR TITLE
Change `min_epsilon` to 0

### LINKED ISSUE #37768: min_epsilon of fused_batch_norm is incorrect
**System information** 
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): n/a
- OS Platform and Distribution (e.g.,Linux Ubuntu 16.04): n/a
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: n/a
- TensorFlow installed from (source orbinary): n/a
- TensorFlow version (use command below):  n/a
- Python version: n/a
- Bazel version (if compiling from source): n/a
- GCC/Compiler version (if compiling from source): n/a
- CUDA/cuDNN version: n/a
- GPU model and memory: n/a

The source code in current master branch has this:
https://github.com/tensorflow/tensorflow/blob/0c853d6cadf055be10562cfec92a93776fafd555/tensorflow/python/ops/nn_impl.py#L1516-L1519

However, the statement in the comment is no longer correct, according to https://docs.nvidia.com/deeplearning/sdk/cudnn-release-notes/rel_750.html:
> the value of epsilon is required to be greater or equal to CUDNN_BN_MIN_EPSILON which was defined in the cudnn.h file to the value 1e-5. This threshold value is now lowered to 0.0 to allow a wider range of epsilon value.

If compatibility with earlier version of cudnn is needed, the minimum should be obtained by using the `CUDNN_BN_MIN_EPSILON` constant defined by `cudnn.h`. 
Or, maybe it's better to just throw an error when a wrong epsilon is provided by the user.  Personally I don't like my parameter to be silently changed -- this can cause issues like `nn.fused_batch_norm` and `nn.batch_normalization` producing different results.

### PR DESCRIPTION
**Pull request by mohantym**
Compiling issue #37768 on epsilon condition as per users input `"the value of epsilon is required to be greater or equal to CUDNN_BN_MIN_EPSILON which was defined in the cudnn.h file to the value 1e-5. This threshold value is now lowered to 0.0 to allow a wider range of epsilon value.` in this [document ](https://docs.nvidia.com/deeplearning/cudnn/release-notes/)

### CODE REVIEW

**Review by mihaimaruseac — APPROVED** [review]

**Comment by ppwwyyxx** [comment]
This is not what the issue meant. 
CuDNN doc says "This threshold value is now lowered to 0.0 to allow a wider range of epsilon value". So the correct fix is to change `min_epsilon` to 0.

File: tensorflow/python/ops/nn_impl.py [hunk 1]
```diff
@@ -1672,8 +1672,9 @@ def fused_batch_norm(
 
   # Set a minimum epsilon to 1.001e-5, which is a requirement by CUDNN to
```
Comment (mihaimaruseac) [inline tensorflow/python/ops/nn_impl.py:1673]: Comment is now wrong.

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/python/ops/nn_impl.py (hunk 1 above)
Comment (mohantym) [reply tensorflow/python/ops/nn_impl.py:1673]: updated comment. Thanks!

File: tensorflow/python/ops/nn_impl.py [hunk 2]
```diff
@@ -1672,8 +1672,9 @@ def fused_batch_norm(
...
   # prevent exception (see cudnn.h).
+
```
Comment (mihaimaruseac) [inline tensorflow/python/ops/nn_impl.py:1675]: Please don't add new empty lines

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]
Now you are no longer following the review from this comment https://github.com/tensorflow/tensorflow/pull/53065#issuecomment-969971493

**Comment by mihaimaruseac** [comment]
Still not following the comment review

**Review by mihaimaruseac — APPROVED** [review]

File: tensorflow/python/eager/backprop.py [hunk 3]
```diff
@@ -806,8 +806,8 @@ class GradientTape(object):
   >>> print(dy_dx)
   tf.Tensor(4.0, shape=(), dtype=float32)
   >>> # No gradients will be available as `w` isn't being watched.
-  >>> dz_dy = tape.gradient(z, w)
-  >>> print(dz_dy)
+  >>> dz_dw = tape.gradient(z, w)
```
Comment (mihaimaruseac) [inline tensorflow/python/eager/backprop.py:809]: This change does not matter

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

**Review by mihaimaruseac — APPROVED** [review]

**Review by gbaned — APPROVED** [review]

**Comment by mohantym** [comment]
Hi @mihaimaruseac ! Shall we close this pull request? it has been going through a lot of loops for a long time now. I am afraid it might disturb the peace of processes happening here.

**Comment by mihaimaruseac** [comment]
I don't have time to see why the internal tests are failing. If you can take a look and fix them that would be great.

**Comment by mihaimaruseac** [comment]
Let's just close this, unless you can check why the internal tests are failing.

**Comment by mohantym** [comment]
@mihaimaruseac !
Agreed ! Closing this PR as internal testing are failing.
Thank you!

**Pull request closed without merging**